src/Compilers/Core/Portable/RuleSet/RuleSet.cs

```diff
@@ -332,7 +332,7 @@ private static ReportDiagnostic GetDiagnosticOptionsFromRulesetFile(Dictionary<s
             }
             catch (IOException e)
             {
-                if (e is FileNotFoundException || e is DirectoryNotFoundException)
+                if (e is FileNotFoundException || e.GetType().Name == "DirectoryNotFoundException")
```

Comment:
@tmat @davkean the intent is for the final state of the code to not have this and to use type comparisons which will be supported in .NET Native (at least to my knowledge).